sdv/tabular/base.py

```diff
@@ -387,8 +418,11 @@ def sample(self, num_rows=None, max_retries=100, max_rows_multiplier=10,
 
         transformed_conditions = self._metadata.transform(conditions, on_missing_column='drop')
         condition_columns = list(conditions.columns)
+        transformed_columns = list(transformed_conditions.columns)
         conditions.index.name = '__condition_idx__'
         conditions.reset_index(inplace=True)
+        transformed_conditions.index.name = '__condition_idx__'
```

Comment:
Since this `'__condition_idx__'` string is now being used in a lot of places, it may be worth to use a module level `CONSTANT` variable instead.
Also, in order to avoid any possible clashes with existing column names (even though `__condition_idx__` is an unlikely string), we can use an `uuid` generated on the fly for it:

```python3
COND_IDX = str(uuid.uuid4())
```